src/OpenLoco/CompanyManager.cpp

```diff
@@ -190,28 +217,61 @@ namespace OpenLoco::CompanyManager
     // Returns a string between 1810 and 1816 with up to two arguments.
     string_id getOwnerStatus(CompanyId_t id, FormatArguments& args)
     {
-        registers regs;
-        regs.esi = X86Pointer(get(id));
-        call(0x00438047, regs);
+        auto& company = _companies[id];
+        if (company.challenge_flags & CompanyFlags::bankrupt)
+            return StringIds::company_status_bankrupt;
+
+        const string_id observationStatusStrings[] = {
+            StringIds::company_status_empty,
+            StringIds::company_status_building_track_road,
+            StringIds::company_status_building_airport,
+            StringIds::company_status_building_dock,
+            StringIds::company_status_checking_services,
+            StringIds::company_status_surveying_landscape,
+        };
 
-        args.push(regs.ecx);
-        args.push(regs.edx);
-        return regs.bx;
-    }
+        string_id statusString = observationStatusStrings[company.observation_status];
+        if (company.observation_town_id == 0xFFFF)
+            return statusString;
 
-    OwnerStatus getOwnerStatus(CompanyId_t id)
```

Comment:
I kind of liked the idea of returning the struct and leaving the caller to pack it into the format arguments if required

Reply:
It was only used in one location, though; the other callers all passed a formatter. I don't think it's likely to pop up elsewhere.